### PR TITLE
publiccloud: propagate retval into LTP_KNOWN_ISSUES flow

### PR DESCRIPTION
**Pull request by asmorodskyi**
This is follow up of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12734 and https://github.com/metan-ucw/runltp-ng/pull/41 . Which fully enable use of LTP_KNOWN_ISSUES in public cloud LTP runs

### CODE REVIEW

**Review by mdoucha — APPROVED** [review]
LGTM

**Review by pevik — APPROVED** [review]
LGTM

File: tests/publiccloud/run_ltp.pm [hunk 1]
```diff
@@ -141,12 +141,14 @@ sub cleanup {
...
-    my $env             = gen_ltp_env();
+    my $content         = Mojo::File::path($ltp_log)->slurp;
+    my $decoded_ltp_log = Mojo::JSON::from_json($content);
+    my $env             = $self->gen_ltp_env();
     my $suite           = get_required_var('COMMAND_FILE');
 
     foreach my $test (@{$decoded_ltp_log->{results}}) {
         if (($test->{status} eq 'fail' || $test->{status} eq 'brok' || $test->{status} eq 'warn')) {
+            $env->{retval} = $test->{retval};
             my $entry = find_whitelist_entry($env, $suite, $test->{test_fqn});
```
Comment (cfconrad) [inline tests/publiccloud/run_ltp.pm:152]:
```suggestion
            $env->{retval} = $test->{retval};
            my $entry = find_whitelist_entry($env, $suite, $test->{test_fqn});
            next unless (defined($entry));
```

Comment (cfconrad) [reply tests/publiccloud/run_ltp.pm:152]: We never check if a whitelist entry exists, so from my perspective we just blindly overwrite all results?!

**Review by cfconrad — CHANGES_REQUESTED** [review]
I have the feeling this isn't read yet.

E.g. `download_whitelist();` doesn't return anything useful and we use it to decide if we wanna `process_ltp_known_issues()`.

**Comment by cfconrad** [comment]
As @asmorodskyi is on holiday, we continue with this: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13065

**Pull request closed without merging**